development-docs/TESTING.md

```diff
@@ -0,0 +1,188 @@
+implicitly stopped because of static block.
+
+Teardown is triggered in `@AfterAll` of `HttpBridgeTestBase`:
+```
+    @AfterAll
+    static void afterAll(VertxTestContext context) {
+        if ("FALSE".equals(BRIDGE_EXTERNAL_ENV)) {                              <--- checking if external Kafka Bridge should run
+            vertx.close(context.succeeding(arg -> context.completeNow()));      <--- closing the vertx instance
+        } else {
+            // if we running external bridge
+            context.completeNow();
+        }
+    }
+```
+
+## Available Test groups
+
+You need to use the `groups` system property in order to execute a group of tests. For example with the following values:
+
+`-Dgroups=httpbridge` — to execute one test group
+`-Dgroups=httpbridge,amqpbridge` — to execute many test groups
+`-Dgroups=all` — to execute all test groups
+
+If `-Dgroups` system property isn't defined, all tests without an explicitly declared test group will be executed.
+The following table shows currently used tags:
+
+| Name               | Description                                                                        |
+| :----------------: | :--------------------------------------------------------------------------------: |
+| httpbridge         | Http Bridge tests, which guarantee, that functionality of Http Bridge is working.  |
+| amqpbridge         | Amqp Bridge tests, which guarantee, that functionality of Amqp Bridge is working.  |
+
+There is also a mvn profile for the main groups - `httpbridge`, `amqpbridge` and `all`, but we suggest to use profile with id `all` (default) and then include or exclude specific groups.
+If you want specify the profile, use the `-P` flag - for example `-Phttpbridge`.
+
+All available test groups are listed in [Constants](https://github.com/strimzi/strimzi-kafka-bridge/blob/master/src/test/java/io/strimzi/kafka/bridge/Constants.java) class.
+
+## Environment variables
+
+System tests can be configured by several environment variables, which are loaded before test execution.
+
+| Name                                   | Description                                                                              | Default                                         |
+| :------------------------------------: | :--------------------------------------------------------------------------------------: | :----------------------------------------------:|
+| KAFKA_EXTERNAL_ENV                     |  Specify if tests should run against in-memory or external Kafka cluster                 | false                                           |
+| BRIDGE_EXTERNAL_ENV                    | Specify if tests should run against in-memory or external Kafka Bridge                   | false                                           |
+
+## Running single test class
+
+Use the `verify` build goal and provide `-Dit.test=TestClassName[#testMethodName]` system property.
```

Comment:
Is this correct? I thought we use only surefire for Bridge tests.

Reply:
when I tried that earlier, it didn't work, but I thought it's just some typo, can you take a look at it @see-quick?

Reply:
Yeah, `test` goal and `-Dtest` parameter should be used here thanks.